backend/src/test/java/com/glebabramov/backend/controller/ResumeControllerTest.java

```diff
@@ -272,6 +274,185 @@ void createResume_shouldReturn201Created_andCreatedResume_ifUserIsAdmin() throws
+		}
+
+		@Test
+		@DirtiesContext
+		@WithMockUser(username = "Test admin", roles = {"ADMIN"})
+		@DisplayName("...should throw 'Unprocessable Entity' (422) if the user is an admin but the resume's to update associated user's previously assigned resume does not exist and thus cannot be unassigned from it")
+		void updateResume_shouldThrow422UnprocessableEntity_ifAssociatedUserPreviouslyAssignedResumeDoesNotExist() throws Exception {
+			// GIVEN
+			MongoUser corruptedUser = new MongoUser("Another-user-ID", testResumeAssignedUser.username(), testResumeAssignedUser.password(), testResumeAssignedUser.role(), "Some-invalid-resume-ID");
+			mongoUserRepository.save(corruptedUser);
+			resumeRepository.save(testResume);
+			mongoUserRepository.save(testResumeAssignedUser);
+			// WHEN
+			mockMvc.perform(put("/api/admin/resumes/update/")
+							.with(csrf())
+							.contentType(MediaType.APPLICATION_JSON)
+							.content("""
+									{
+										"id": "Some-resume-ID",
+										"name": "Resume name",
+										"userIds": ["Some-user-ID", "Another-user-ID"]
+									}
+									"""))
+					.andExpect(status().isUnprocessableEntity());
+		}
+
+		@Test
+		@DirtiesContext
+		@WithMockUser(username = "Test admin", roles = {"ADMIN"})
+		@DisplayName("...should throw 'Unprocessable Entity' (422) if the user is an admin and the resume's to update associated user's previously assigned resume does exist but the standard resume does not and thus the user cannot be re-assigned to it")
+		void updateResume_shouldThrow422UnprocessableEntity_ifAssociatedUserPreviouslyAssignedResumeExistsButReplacingResumeDoesNotExist() throws Exception {
+			// GIVEN
+			MongoUser anotherTestUser = new MongoUser("Another-user-ID", testResumeAssignedUser.username(), testResumeAssignedUser.password(), testResumeAssignedUser.role(), "Some-valid-resume-ID");
+			mongoUserRepository.save(anotherTestUser);
+			Resume testResume2 = new Resume("Some-valid-resume-ID", testResume.name(), Set.of("Another-user-ID"), testResume.invitationSent(), testResume.isPublished());
+			resumeRepository.save(testResume);
+			resumeRepository.save(testResume2);
+			mongoUserRepository.save(testResumeAssignedUser);
+			// WHEN
+			mockMvc.perform(put("/api/admin/resumes/update/")
+							.with(csrf())
+							.contentType(MediaType.APPLICATION_JSON)
+							.content("""
+									{
+										"id": "Some-resume-ID",
+										"name": "Resume name",
+										"userIds": ["Another-user-ID"]
+									}
+									"""))
+					.andExpect(status().isUnprocessableEntity());
+		}
+
+		@Test
+		@DirtiesContext
+		@WithMockUser(username = "Test admin", roles = {"ADMIN"})
+		@DisplayName("...should update the resume, and handle the associated users' (former and newly set) resume assignments, return 200 (OK) and the updated resume if the user is an admin, resume to update exists, associated users exist and their assigned resumes as well as the replacing resume exist")
```

Comment:
I get a knot in my brain from reading this. I think the sentence structure has some logical issues.